internal/plugin/github.go

```diff
@@ -5,22 +5,45 @@ import (
 	"io"
 	"net/http"
 	"net/url"
+	"regexp"
+	"strings"
 
+	"github.com/samber/lo"
 	"go.jetpack.io/devbox/internal/boxcli/usererr"
 	"go.jetpack.io/devbox/internal/cachehash"
 	"go.jetpack.io/devbox/nix/flake"
 )
 
 type githubPlugin struct {
-	ref flake.Ref
+	ref  flake.Ref
+	name string
+}
+
+// Github only allows alphanumeric, hyphen, underscore, and period in repo names.
+// but we clean up just in case.
+var githubNameRegexp = regexp.MustCompile("[^a-zA-Z0-9-_.]+")
+
+func newGithubPlugin(ref flake.Ref) *githubPlugin {
+	plugin := &githubPlugin{ref: ref}
+	// For backward compatibility, we don't strictly require name to be present
+	// in github plugins. If it's missing, we just use the directory as the name.
+	name, _ := getPluginNameFromContent(plugin)
+	if name == "" {
+		name = strings.ReplaceAll(ref.Dir, "/", "-")
```

Comment:
what happens if the plugin's devbox.json config is in the root of the github repo? Can we default it to "root-folder" or something?

Reply:
If there's no name and it's in the root, the qualified name becomes `owner.repo`